### ISSUE TITLE
aws-sdk SQS autoinstrumentation links not working

### ISSUE THREAD

**Issue by ImDevinC** [open-telemetry/opentelemetry-js-contrib#1814]
### What version of OpenTelemetry are you using?
```json
    "@opentelemetry/api": "^1.7.0",
    "@opentelemetry/auto-instrumentations-node": "^0.40.0",
    "@opentelemetry/exporter-trace-otlp-proto": "^0.45.1",
    "@opentelemetry/sdk-node": "^0.45.1",
    "@opentelemetry/sdk-trace-base": "^1.18.1",
    "@opentelemetry/sdk-trace-node": "^1.18.1",
```

### What version of Node are you using?
`v18.16.1`

### What did you do?
Full example here: https://github.com/ImDevinC/tracing-example
When using the below code sample to read off of an SQS queue, the resulting traces do not have a link to the originating trace. However if I use the `sqs-consumer` package to read off of the queue, then the traces are properly instrumented with a link to the originating trace.
**consume.ts**
```typescript
import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
import {
  ReceiveMessageCommand,
  Message,
  SQSClient,
  DeleteMessageCommand,
  GetQueueUrlCommand,
} from "@aws-sdk/client-sqs";
import { SpanStatusCode, trace } from "@opentelemetry/api";

const tracer = trace.getTracer(process.env.SERVICE_NAME!);

const start = async () => {
  return tracer.startActiveSpan("consume", async (span) => {
    try {
      await consume();
    } catch (err: any) {
      console.log(err);
      span.recordException(err);
      span.setStatus({
        code: SpanStatusCode.ERROR,
        message: JSON.stringify(err),
      });
    }
    span.end();
  });
};

const consume = async () => {
  const sqs = new SQSClient({});
  const publishCommand = new ReceiveMessageCommand({
    QueueUrl: process.env.QUEUE_URL,
  });

  let messages: Message[] = [];

  try {
    const response = await sqs.send(publishCommand);
    messages = response.Messages || [];
  } catch (err) {
    throw err;
  }

  if (messages.length == 0) {
    console.log("No messages found");
    return;
  }

  const s3 = new S3Client({});
  for (const message of messages) {
    console.log(JSON.stringify(message));
    const putCommand = new PutObjectCommand({
      Bucket: process.env.BUCKET_NAME,
      Key: "message.txt",
      Body: message.Body,
    });
    const deleteMessageCommand = new DeleteMessageCommand({
      QueueUrl: process.env.QUEUE_URL,
      ReceiptHandle: message.ReceiptHandle,
    });
    try {
      await s3.send(putCommand);
      await sqs.send(deleteMessageCommand);
    } catch (err) {
      throw err;
    }
  }
};

start().then(async () => {
  return new Promise((resolve) => {
    setTimeout(resolve, 5000);
  }).then(() => {
    console.log("done");
  });
});
```
**telemetry.js**
```javascript
const {
  getNodeAutoInstrumentations,
} = require("@opentelemetry/auto-instrumentations-node");
const {
  OTLPTraceExporter,
} = require("@opentelemetry/exporter-trace-otlp-proto");
const { NodeSDK } = require("@opentelemetry/sdk-node");
const { BatchSpanProcessor } = require("@opentelemetry/sdk-trace-base");
const { ConsoleSpanExporter } = require("@opentelemetry/sdk-trace-node");

// const consoleExporter = new ConsoleSpanExporter();

const exporter = new OTLPTraceExporter({
  url:
    "http://" +
    (process.env.OTEL_COLLECTOR_URL ?? "localhost:4318") +
    "/v1/traces",
});

const sdk = new NodeSDK({
  spanProcessor: new BatchSpanProcessor(exporter),
  serviceName: process.env.SERVICE_NAME,
  instrumentations: [
    getNodeAutoInstrumentations({
      "@opentelemetry/instrumentation-fs": {
        enabled: false,
      },
      "@opentelemetry/instrumentation-aws-sdk": {
        sqsExtractContextPropagationFromPayload: true,
        suppressInternalInstrumentation: true,
      }
    }),
  ],
});

sdk.start();

process.on("SIGTERM", () => {
  sdk
    .shutdown()
    .then(() => console.log("Tracing terminated"))
    .catch((error) => console.log("Error terminating tracing", error))
    .finally(() => process.exit(0));
});
```

### What did you expect to see?
I expect to see a link to the original trace

### What did you see instead?
No link is present

### Additional context
If I instead use the `sqs-consumer` package to read from the queue, then the trace is properly linked. The README in the linked repository outlines the different scenarios.
https://github.com/ImDevinC/tracing-example

**Comment by ImDevinC** [comment]
After digging a bit more, it looks like I'm not getting a `process` span when working on my queue, but I'm not sure why.

**Comment by ImDevinC** [comment]
Ok, turns out reading the docs is very helpful, posting what I found here in case anyone stumbles across this in the future: https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-aws-sdk/doc/sqs.md#processing-spans

tl;dr:
Using the format of 
```typescript
for (const message of messages) { // Do the stuff }
```
Loses the context of the span (which makes sense). After I refactored my code to do something like the following, it works much better:
```typescript
const promises: Promise<void>[] = [];
  messages.forEach((message) => {
    promises.push(processMessage(message));
  });
await Promise.all(promises)

const processMessage = async (message: Message) => { // Do the stuff }
```

If I could use the earlier format, that would be great, but I understand that's not currently supported. Closing this for now.
